llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -15533,8 +15534,16 @@ SDValue DAGCombiner::visitFREEZE(SDNode *N) {
     if (Op.getOpcode() == ISD::UNDEF)
       Op = DAG.getFreeze(Op);
   }
-  // NOTE: this strips poison generating flags.
-  SDValue R = DAG.getNode(N0.getOpcode(), SDLoc(N0), N0->getVTList(), Ops);
+
+  SDValue R;
+  if (auto *SVN = dyn_cast<ShuffleVectorSDNode>(N0)) {
```

Comment:
Wouldn't the else part work just fine also for VECTOR_SHUFFLE?
The mask is in Ops[2].

Reply:
ShuffleVectorSDNode embeds the mask in the node itself - its not an operand

Reply:
Ok. Got it.